### PR TITLE
[TE] Self-Serve Improvements (graph and email validation)

### PR DESCRIPTION
**Pull request by newsummit**
- fixed graph legend issues (only show legend on left if graphing dimensional data)
- fixed issue where legend was not showing labels for alert-edit
- improved email validation in edit view, added it to create view

### CODE REVIEW

File: thirdeye/thirdeye-frontend/app/pods/manage/alerts/edit/controller.js [hunk 1]
```diff
@@ -127,6 +127,15 @@ export default Ember.Controller.extend({
   ),
 
   /**
+   * We only want to display the custom legend if the alert has a value for dimension exploration
+   * @method showGraphLegend
+   * @return {Boolean}
+   */
+  showGraphLegend: Ember.computed('alertDimension', function() {
```
Comment (justYves) [inline thirdeye/thirdeye-frontend/app/pods/manage/alerts/edit/controller.js]: fyi: I think this should work also 
```
showGraphLegend: Ember.computed.notEmpty('alertDimension')
```
https://www.emberjs.com/api/ember/2.15/namespaces/Ember.computed/methods/bool?anchor=notEmpty

File: thirdeye/thirdeye-frontend/app/pods/manage/alerts/edit/controller.js [hunk 2]
```diff
@@ -245,7 +254,10 @@ export default Ember.Controller.extend({
     'isDuplicateEmail',
     'isProcessingForm',
     function() {
-      return this.get('isEmptyEmail') || this.get('isProcessingForm') || this.get('isEmailError' || this.get('isDuplicateEmail'));
```
Comment (justYves) [inline thirdeye/thirdeye-frontend/app/pods/manage/alerts/edit/controller.js]: fyi: this should be shorter to write: 
```
isSubmitDisabled: Ember.computed.or('{isEmptyEmail, isEmailError, isDuplicateEmail ,isProcessingForm}'),
```
https://www.emberjs.com/api/ember/2.15/namespaces/Ember.computed/methods/bool?anchor=or

File: thirdeye/thirdeye-frontend/app/pods/manage/alerts/edit/controller.js [hunk 3]
```diff
@@ -320,9 +332,13 @@ export default Ember.Controller.extend({
   isEmailValid(emailArr) {
     const emailRegex = /^.{3,}\@linkedin.com$/;
     let isValid = true;
+
     for (var email of emailArr) {
```
Comment (justYves) [inline thirdeye/thirdeye-frontend/app/pods/manage/alerts/edit/controller.js]: nit: this looks like a good candidate for a `reduce` (m to 1)

File: thirdeye/thirdeye-frontend/app/pods/manage/alerts/edit/controller.js [hunk 4]
```diff
@@ -336,7 +352,7 @@ export default Ember.Controller.extend({
     this.set('isEditAlertSuccess', true);
     Ember.run.later((function() {
       that.clearAll();
-      this.transitionToRoute('manage.alerts');
+      that.transitionToRoute('manage.alerts');
```
Comment (justYves) [inline thirdeye/thirdeye-frontend/app/pods/manage/alerts/edit/controller.js]: ### Blocking comment: 
let's refrain from using `that` . The `Ember.run.later` api allows to pass the context before the callback. 
https://www.emberjs.com/api/ember/2.15/namespaces/Ember.computed/methods/bool?anchor=notEmpty

```
Ember.run.later(this, (function() {
})
```

File: thirdeye/thirdeye-frontend/app/pods/self-serve/create-alert/controller.js [hunk 5]
```diff
@@ -1064,13 +1118,29 @@ export default Ember.Controller.extend({
 
       // This object contains the data for the new alert function, with default fillers
       const newFunctionObj = this.get('newAlertProperties');
```
Comment (justYves) [inline thirdeye/thirdeye-frontend/app/pods/self-serve/create-alert/controller.js]: nit: let's use the destructuring here: 
```
const {newFunctionObj, oldEmails,  newEmails} = this.getProperties('....')
```

File: thirdeye/thirdeye-frontend/app/pods/manage/alerts/edit/controller.js (hunk 1 above)
Comment (newsummit) [reply thirdeye/thirdeye-frontend/app/pods/manage/alerts/edit/controller.js]: nice - thanks.

File: thirdeye/thirdeye-frontend/app/pods/manage/alerts/edit/controller.js (hunk 2 above)
Comment (newsummit) [reply thirdeye/thirdeye-frontend/app/pods/manage/alerts/edit/controller.js]: thanks

File: thirdeye/thirdeye-frontend/app/pods/manage/alerts/edit/controller.js (hunk 3 above)
Comment (newsummit) [reply thirdeye/thirdeye-frontend/app/pods/manage/alerts/edit/controller.js]: actually I will use `emailArr.every(email => emailRegex.test(email))`

File: thirdeye/thirdeye-frontend/app/pods/manage/alerts/edit/controller.js (hunk 4 above)
Comment (newsummit) [reply thirdeye/thirdeye-frontend/app/pods/manage/alerts/edit/controller.js]: Good to know!

**Comment by newsummit** [comment]
Addressed recommendations + added a fix for finding application name in search results.

**Pull request merged**